lib/open_api/serializer.rb

```diff
@@ -1,14 +1,43 @@
 module OpenApi
   module Serializer
     def as_json(arg = {})
-      return super unless kind_of?(ApplicationRecord)
+      previous = super
+      encrypted_columns_set = (self.class.try(:encrypted_columns) || []).to_set
+      encryption_filtered = previous.except(*encrypted_columns_set)
+      version = version_from_path(arg)
 
-      super.tap do |hash|
-        hash.each_key do |attr_key|
-          hash[attr_key] = hash[attr_key].iso8601 if hash[attr_key].kind_of?(Time)
-          hash[attr_key] = hash[attr_key].to_s if attr_key.ends_with?("_id") || attr_key == "id"
-        end
+      return encryption_filtered unless arg.key?(:prefixes) || version
+
+      # We use "#{klass.name}Out" for the definitions on what are returned.
+      schema = Api::Docs[version].definitions[self.class.name + "Out"]
+
+      filter(encryption_filtered.slice(*schema["properties"].keys), schema, encrypted_columns_set)
+    end
+
+    private
+
+    def version_from_path(arg)
+      url = ManageIQ::API::Common::Request.current.instance_variable_get(:@original_url) || nil
+      base_path = url ? URI.parse(url).path : arg[:prefixes].first
+      api_version_from_prefix(base_path)
+    end
+
+    def filter(attrs, schema, encrypted_columns_set)
+      schema["properties"].keys.each do |name|
+        next if attrs[name].nil?
+
+        attrs[name] = attrs[name].iso8601 if attrs[name].kind_of?(Time)
+        attrs[name] = attrs[name].to_s if name.ends_with?("_id") || name == "id"
+        attrs[name] = public_send(name) if !attrs.key?(name) && !encrypted_columns_set.include?(name)
       end
+      attrs.compact
+    end
+
+    def api_version_from_prefix(prefix)
+      return unless prefix
```

Comment:
protecting this line does not prevent method `as_json` from crashing. you can just drop it.

Reply:
@bzwei to give some context here.  The original serializer code in common is here.  https://github.com/ManageIQ/manageiq-api-common/blob/master/lib/manageiq/api/common/open_api/serializer.rb

I would like to pull all the `prefix` checking out. It appears it was added to help test the serializer more than anything.  I left it in on Catalog to help possibly ease our ability to use the manageiq-api-common version with the least amount of changes.